### PR TITLE
Author qualifier does not consider real name #1489

### LINKED ISSUE #1489: Author qualifier does not consider real name

### PR DESCRIPTION
**Pull request by raisfathin**
Fixes #1489

### CODE REVIEW

**Comment by raisfathin** [comment]
@dariusf currently, this PR assumes that in the future, the real name of non collaborators will be stored inside `Model`. There is another alternative which creates a `TurboUser` field inside `TurboIssue` to store the information of the creator instead of just their username.

Which one do you think is better?

**Comment by dariusf** [comment]
The first one. The second will duplicate information, since TurboIssue is meant to be serialised.

The way we store data is (kind of) relational -- we store keys instead of nesting objects. It isn't the most pleasant to use, as we have to join the data explicitly now, and this should ideally be done behind the scenes. Hopefully we can look into improving the interface soon.

**Comment by raisfathin** [comment]
I see. Is there anything else that I should add besides looking for author in `Model`?

**Comment by dariusf** [comment]
Don't think so, at least for this issue.

File: src/main/java/filter/expression/Qualifier.java [hunk 1]
```diff
@@ -873,15 +873,23 @@ private boolean assigneeSatisfies(IModel model, TurboIssue issue) {
...
 
-        String creator = issue.getCreator();
-        return creator.toLowerCase().contains(content.get().toLowerCase());
+        String content = this.content.get().toLowerCase();
+        String login = issue.getCreator().toLowerCase();
+        String name = ""; // default value, since TurboIssue does not keep track of creator's real name
+
+        if (author.isPresent()) {
+            name = author.get().getRealName() == null ? "" : author.get().getRealName().toLowerCase();
+        }
```
Comment (dariusf) [inline src/main/java/filter/expression/Qualifier.java]: We can simplify this part a bit:

```java
String name = author.map(TurboUser::getRealName)
    .orElse("").toLowerCase();
```

**Comment by raisfathin** [comment]
Done! Ready for review.

**Pull request merged**